client/src/components/Flights/FlightController.jsx

```diff
@@ -28,6 +28,7 @@ export default function FlightController({ setFlights, nonStop, setNonStop, roun
     })
       .then((response) => {
         setFlights(response.data.data);
+        setLoading(false);
       })
       .catch((err) => {
         console.log(err);
```

Comment:
Should probably set loading to false here as well since we have received a response